### PR TITLE
Be able to execute bash action from file

### LINKED ISSUE #107: Execute bash action from file
Allow the execution of a bash action by loading an external sh file instead of pasting the content into the bash building block

### PR DESCRIPTION
**Pull request by antoinecarton**
closes #107

### CODE REVIEW

File: engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt [hunk 1]
```diff
@@ -36,6 +37,14 @@ fun ChangeSetDsl.bashAction(
     }
 }
 
+fun ChangeSetDsl.bashAction(
+    workingDir: String = ".",
+    timeout: Duration = Duration.ofMinutes(1),
+    file: File
```
Comment (ilaborie) [inline engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt:43]: IMO could be the first argument

Comment (ilaborie) [reply engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt:43]: Question: the `workingDir` may be by default on the file dir ?

File: samples/bash/src/main/resources/changelog.kts [hunk 2]
```diff
@@ -27,5 +30,6 @@ changeLog {
                     export | grep " PWD"
                 """.trimIndent()
         }
+        bashAction(workingDir = "/tmp", file = File(Main::class.java.getResource("pwet.sh").toURI()))
```
Comment (ilaborie) [inline samples/bash/src/main/resources/changelog.kts:33]: Note: `File(Main::class.java.getResource(...).toURI())` could fail

Comment (antoinecarton) [reply samples/bash/src/main/resources/changelog.kts:33]: yes, it therefore sets the changeSet to fail. This is the wanted behavior?

File: engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt (hunk 1 above)
Comment (antoinecarton) [reply engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt:43]: We'll move it to the first argument :+1: 
The working dir is the directory where the script is run and may be different from the one where the script is located

Comment (antoinecarton) [reply engine-bash/src/main/kotlin/io/saagie/updatarium/model/action/BashScriptAction.kt]: done

File: samples/bash/src/main/resources/changelog.kts [hunk 3]
```diff
@@ -27,5 +28,6 @@ changeLog {
                     export | grep " PWD"
                 """.trimIndent()
         }
+        bashAction(File(Main::class.java.getResource("pwet.sh").file))
```
Comment (ilaborie) [inline samples/bash/src/main/resources/changelog.kts:31]: I think you should avoid the `.getResource(...).file`, and just use a String instead.
But, that's a minor remark

**Comment by PierreLeresteux** [comment]
Donc forget to update documentation to explain this feature !

**Pull request merged**